nixos/modules/system/activation/switchable-system.nix

```diff
@@ -36,6 +42,14 @@ in
 
   config = lib.mkMerge [
     (lib.mkIf (config.system.switch.enable && !config.system.switch.enableNg) {
+      warnings = [
+        ''
+          The Perl implementation of switch-to-configuration will be deprecated
+          and removed in a future release of NixOS. Please migrate to the newer
+          implementation by setting `system.switch.enableNg = true`.
```

Comment:
Might be good to:

* explicitly mention 25.05 as the release it’s likely to be removed in (so people know it’s coming imminently);

* tell people to report an issue if they have issues with the Rust rewrite that stop them from using it (and possibly tag you?).

Reply:
done!

Reply:
Still hoping for a release note change :) Also it might be a good idea to just tell people to *remove* their `enableNg` setting, as we’d hope to remove the option too and it can only be `false` if people are explicitly setting it to be so.

Reply:
Got it, should be changed now! LMK if the wording doesn't sound right